src/ServiceContext.cpp

```diff
@@ -31,6 +31,12 @@ std::string ServiceContext::getServerIpFromClientIp(uint32_t client_ip) const
     char server_ip[INET_ADDRSTRLEN];
 
 
+    if (eth_ifs.size() == 1)
```

Comment:
My solution is not smart but solves the problem explained in thie issue:
https://github.com/roleoroleo/yi-hack-MStar/issues/148
When the client comes from a different subnet, the server returns localhost.

Reply:
But this will not work if the size is greater than 1. The solution does not seem correct. And I don't understand how a connection to a server from a different subnet can be obtained, if there is another subnet, it needs to be added as an alias, say eth0:1. The subnet routing issue should be at the router level.

Reply:
Right: this will not work if the size is geater than 1, but I don't know how to get the destination IP address when there are 2 or more network adapters through soap.

If you have the client in the subnet 192.168.0.x and the server in the subnet 192.168.1.x the following condition is always false:
```
if( (if_ip & if_mask) == (client_ip & if_mask) )
```
i.e.
if_ip = 192.168.1.100 and client_ip = 192.168.0.1

So the function will return 127.0.0.1.

Reply:
Have you used netplan for network configuration? because it doesn't create aliases like eth0:0 eth0:1 etc.

Reply:
I understood the reason for your decision. I think I know how to write generic code.